sympy/combinatorics/perm_groups.py

```diff
@@ -3815,7 +3815,301 @@ def _sylow_reduce(mu, nu):
+                rels.append(new_rel)
+
+        return True, rels
+
+    def strong_presentation(G):
+        '''
+        Return a strong finite presentation of `G`. The generators
+        of the returned group are in the same order as the strong
+        generators of `G`.
+
+        The algorithm is based on Sims' Verify algorithm described
+        in [1], Chapter 6.
+
+        Example
+        =======
+        >>> from sympy.combinatorics.perm_groups import PermutationGroup
+        >>> from sympy.combinatorics.named_groups import DihedralGroup
+        >>> P = DihedralGroup(4)
+        >>> G = P.strong_presentation()
+        >>> P.order() == G.order()
+        True
+
+        See Also
+        ========
+        presentation, _verify
+
+        '''
+        from sympy.combinatorics.fp_groups import (FpGroup,
+                                            simplify_presentation)
+        from sympy.combinatorics.free_groups import free_group
+        from sympy.combinatorics.homomorphisms import (block_homomorphism,
+                                           homomorphism, GroupHomomorphism)
+        from itertools import product
+
+        strong_gens = G.strong_gens[:]
+        stabs = G.basic_stabilizers[:]
+        base = G.base[:]
+
+        # injection from a free group on len(strong_gens)
+        # generators into G
+        gen_syms = [('x_%d'%i) for i in range(len(strong_gens))]
+        F = free_group(', '.join(gen_syms))[0]
+        phi = homomorphism(F, G, F.generators, strong_gens)
+
+        H = PermutationGroup(G.identity)
+        while stabs:
+            alpha = base.pop()
+            K = H
+            H = stabs.pop()
+            new_gens = [g for g in H.generators if g not in K]
+
+            if K.order() == 1:
+                z = new_gens.pop()
+                rels = [F.generators[-1]**z.order()]
+                intermediate_gens = [z]
+                K = PermutationGroup(intermediate_gens)
+
+            # add generators one at a time building up from K to H
+            while new_gens:
+                z = new_gens.pop()
+                intermediate_gens = [z] + intermediate_gens
+                K_s = PermutationGroup(intermediate_gens)
+                orbit = K_s.orbit(alpha)
+                orbit_k = K.orbit(alpha)
+
+                # split into cases based on the orbit of K_s
+                if orbit_k == orbit:
+                    if z in K:
+                        rel = phi.invert(z)
+                        perm = z
+                    else:
+                        t = K.orbit_rep(alpha, alpha^z)
+                        rel = phi.invert(z)*phi.invert(t)**-1
+                        perm = z*t**-1
+                    for g in K.generator_product(perm, original=True):
+                        rel = rel*phi.invert(g)**-1
+                    new_rels = [rel]
+                elif len(orbit_k) == 1:
+                    success, new_rels = K_s._verify(K, phi, z, alpha)
```

Comment:
Shouldn't `success` be checked somewhere?

Reply:
The way the function is written, we know that `success` will always be true because we start with a strong generating set. Theoretically, later this could be changed so that we start with a randomly generated (potential) BSGS, and then append new elements to it whenever `success` is false.

Reply:
OK. Maybe a comment could be added to that effect for future maintenance.

Reply:
Okay, will add one.

Reply:
Otherwise, I think this is ready to be merged.